simplecnf.go

```diff
@@ -0,0 +1,160 @@
+package go_sat_solver
+
+import "math"
+
+type orClause []VariableAssignment
+
+type simpleSatCnf struct {
+	clauses []orClause
+}
+
+func (f simpleSatCnf) IsSatisfied() bool {
+	return len(f.clauses) == 0
+}
+
+func (f simpleSatCnf) IsContradiction() bool {
+	for _, c := range f.clauses {
+		if len(c) == 0 {
+			return true
+		}
+	}
+	return false
+}
+
+type variableAssignment map[Variable]bool
```

Comment:
this is an acceptable name, but it makes the other name completely unacceptable. change one of these. preferably the other one.